### DISCUSSION TITLE
What do the different lock icons mean?

### DISCUSSION THREAD

**Discussion by qubitz** [spoiledcat/git-for-unity#10] [Q&A] [answered]
When starting to play around with locks yesterday, I updated the .gitattributes files to support locking particular files assuming it was a requirement.
```diff
* text=auto
+ *.psd filter=lfs diff=lfs merge=lfs -text lockable
```

Surprisingly, it doesn't seem like a requirement. I was very excited to see how easy it was to lock files and so I tried locking a file that wasn't annotated with `lockable` within the .gitattributes and the file _still_ locked. 

Getting to the point, when pulling up the same project on another machine, the locked files had different colored icons. Is this because one of the files is `lockable` and other is not?
![image](https://user-images.githubusercontent.com/15070724/203447344-b3e5a2b9-c348-49a7-a737-45d62f1bae45.png)

**Comment by CapnRat** [comment, marked as answer]
Both icons mean that the file is locked. The purple one (without the person shape) shows up when the file also has local modifications.

Regarding the `lockable` gitattribute. git lfs allows locking of any file. The lockable attribute purpose is to make locked files read-only locally. (See https://github.com/git-lfs/git-lfs/issues/3365#issuecomment-436755372 ) However, with Unity that doesn't matter much since Unity itself forces all files that it wants to write to as writeable. Unfortunately, not within the power of packages to change that behavior.
